src/transformers/models/speecht5/number_normalizer.py

```diff
@@ -0,0 +1,198 @@
+                    if tens_units > 0:
+                        part += " and "
+
+                if tens_units > 10 and tens_units < 20:
+                    part += self.teens[tens_units - 10]
+                else:
+                    tens_digit = self.tens[tens_units // 10]
+                    ones_digit = self.ones[tens_units % 10]
+                    if tens_digit:
+                        part += tens_digit
+                    if ones_digit:
+                        if tens_digit:
+                            part += " "
+                        part += ones_digit
+
+                parts.append(part)
+
+            num //= 1000
+
+        return " ".join(reversed(parts))
+
+    def convert(self, n):
+        if "." in n:
+            integer_part, decimal_part = n.split(".")
+        else:
+            integer_part, decimal_part = n, "00"
+
+        # Define a dictionary to map currency symbols to their names
+        currency_symbols = {
+            "$": " dollars",
+            "€": " euros",
+            "£": " pounds",
+            "¢": " cents",
+            "¥": "Japanese Yen",
+            "₹": "Indian Rupee",
+            "₽": "Russian Ruble",
+            "฿": "Thai Baht",
+            "₺": "Turkish Lira",
+            "₴": "Ukrainian Hryvnia",
+            "₣": "Swiss Franc",
+            "₡": "Costa Rican Colon",
+            "₱": "Philippine Peso",
+            "₪": "Israeli New Shekel",
+            "₮": "Mongolian Tögrög",
+            "₩": "South Korean Won",
+            "₦": "Nigerian Naira",
+            "₫": "Vietnamese Đồng",
+        }
+
+        # Extract currency symbol if present
+        currency_symbol = ""
+        for symbol, name in currency_symbols.items():
+            if integer_part.startswith(symbol):
+                currency_symbol = name
+                integer_part = integer_part[len(symbol) :]
+                break
+
+            if integer_part.startswith("-"):
+                if integer_part[1:].startswith(symbol):
+                    currency_symbol = name
+                    integer_part = "-" + integer_part[len(symbol) + 1 :]
+                    break
+
+        # Extract 'minus' prefix for negative numbers
+        minus_prefix = ""
+        if integer_part.startswith("-"):
+            minus_prefix = "minus "
+            integer_part = integer_part[1:]
+        elif integer_part.startswith("minus"):
+            minus_prefix = "minus "
+            integer_part = integer_part[len("minus") :]
+
+        percent_suffix = ""
+        if "%" in integer_part or "%" in decimal_part:
+            percent_suffix = " percent"
+            integer_part = integer_part.replace("%", "")
+            decimal_part = decimal_part.replace("%", "")
+
+        print(integer_part, decimal_part)
+
+        # integer_part = re.sub(r"\D", "", integer_part)
+        integer_part = integer_part.zfill(3 * ((len(integer_part) - 1) // 3 + 1))
+
+        parts = []
+        units = [
```

Comment:
Can we not use `self.thousands` here?

Reply:
Yes! I actually wrote this function separately while working on it so I didn't integrate it well with the class. Fixed now!